src/posts/2022-11-27-define-once/define-once.md

```diff
@@ -0,0 +1,355 @@
+---
+title: Define a python function once and for all
+subtitle: A terrible hack that ends up being surprisingly useful
+---
+
+Sometimes when we have a function that takes a long time to run, and we would
+really rather avoid running the function more often than absolutely necessary.
+Luckily, if our function is a pure function[^1], then we can
+[memoize](https://en.wikipedia.org/wiki/Memoization) the result of calling the
+function using the `cache` function.
```

Comment:
Maybe just "using `cache`." to avoid using the word "function" so many times in one sentence.

Reply:
Done! Reduced the many uses of the word "function" to just one use.